superset-frontend/packages/superset-ui-core/src/components/Dropdown/index.tsx

```diff
@@ -110,8 +110,61 @@ export const NoAnimationDropdown = (props: NoAnimationDropdownProps) => {
   );
 };
 
+const CustomDropdownButton = (props: ButtonProps & DropdownProps) => {
+  const theme = useTheme();
+  const {
+    // Dropdown-specific props
+    overlay,
+    placement = 'bottomRight',
+
+    // Button-specific props
+    children,
+    buttonStyle,
+    disabled,
+    ...rest
+  } = props;
+
+  return (
+    <Space.Compact
```

Comment:
as a note, I think we'll try to avoid doing things like this in component library, the spacing / layout of the component should be left to the caller, unless it's super clear that we always need spacing to be a certain way. If anything we should create or use things like a `ButtonGroup` or things like `ButtonSpace` if/when things get repetitive.

Keeping antd as vanilla as possible allows for working with the assumption that you can assume things are "generic antd", so or most of the antd docs, or AI-generate code based on AI's understanding of antd is accurate. "Oddly mutated components from antd" is probably confusing to everyone, people and AIs